### ISSUE TITLE
Gui

### ISSUE THREAD

**Issue by tobyplowy** [D00Med/farlands#87]
@GreenXenith this is your responsibility 

What needs to be done 

Top prioritie
Survival inventory
Creative inventory

Low priority
Furnace
Crafting table
Anvil if it gets added (most likely yes)
Npc (villager) trading gui

**Comment by GreenXenith** [comment]
**Player Menus**
- [x] Survival Slots
- [ ] Survival Tabs
  - [ ] Armor
  - [ ] Craft Guide
  - [ ] Journal
  - [ ] I think I am forgetting something
- [ ] Creative Slots
- [ ] Creative Tabs
  - [ ] Creative Menu
  - [ ] Armor
  - [ ] Craft Guide
  - [ ] Journal
  - [ ] I think I am forgetting something

**Nodes**
- [x] Chest Formspec
- [x] Large Chest
  - [x] Large Chest Node
  - [x] Large Chest Formspec
- [ ] Furnace Formspec

**Comment by tobyplowy** [comment]
@GreenXenith thanks for updating ;D

**Comment by tobyplowy** [comment]
> I think I am forgetting something

Maybe Achievements?
Apart from that I don't think you did

**Comment by tobyplowy** [comment]
@GreenXenith can you also make the repair tool function on the crafting table a separate block? (an anvile)

you dount have too im just asking

**Comment by tobyplowy** [comment]
@GreenXenith crafting table concept
![crafting](https://cloud.githubusercontent.com/assets/16853304/26416966/ff3e8dda-40b7-11e7-8a4c-bac04bc18c45.png)

**Comment by GreenXenith** [comment]
@tobyplowy So I am putting a craft guide on the table, not the inventory?

**Comment by tobyplowy** [comment]
@GreenXenith No both Because of you use the crafting table gui you won't be able to access your own inventory unless You Go Out Of The Crafting gui and Open Your inventory (same for the other way around)

**Comment by tobyplowy** [comment]
@GreenXenith On The Other hand in the players Crafting gui you won't really Need a recipe book because the only recipe that's important there is a 2 by 2 wood craft recipe to make a crafting table (and some other basic stuff)

It can also be an item (crafting recipe book item) that maybe better but you can choose

**Comment by GreenXenith** [comment]
I think I might just do the book.

**Comment by tobyplowy** [comment]
Ok

**Comment by tobyplowy** [comment]
@GreenXenith I will make a recipe book texture (I'm working on a book texture at the moment anyway )

**Comment by tobyplowy** [comment]
craft book
![craft book](https://cloud.githubusercontent.com/assets/16853304/26426110/83631698-40d7-11e7-9f44-125313f312ee.png)

**Comment by GreenXenith** [comment]
@tobyplowy That is 16x17. Is it possible to make it 16x16?

**Comment by tobyplowy** [comment]
@GreenXenith oh how did that happen :/

**Comment by tobyplowy** [comment]
![craft book](https://cloud.githubusercontent.com/assets/16853304/26426423/cc3c1e54-40d8-11e7-9a8c-c53970d5af4e.png)

**Comment by GreenXenith** [comment]
ty :)

**Comment by tobyplowy** [comment]
the one in minetest game is 16x17 st thats why that happend
https://github.com/minetest/minetest_game/blob/master/mods/default/textures/default_book.png

**Comment by DS-Minetest** [comment]
Hm, many people would like a new sort of crafting system, something like this: https://forum.minetest.net/viewtopic.php?t=15405
If farlands would have something like that, it would differ more from the other subgames.

**Comment by tobyplowy** [comment]
> https://forum.minetest.net/viewtopic.php?t=15405

I don't like that I prefer it the good ol way XD

**Comment by GreenXenith** [comment]
> I prefer it the good ol way

Agreed.

**Comment by DS-Minetest** [comment]
It's just the normal way of minetest. Other games do it like this, eg. terraria.

IMO the style of the textures of the gui are too gray and sterile.

**Comment by tobyplowy** [comment]
@DS-Minetest yes I will improve the color but first I wanna see the GUI finished

**Comment by GreenXenith** [comment]
If you plan on modifying any texture, do **NOT** change image sizes or slot positions.

**Comment by tobyplowy** [comment]
@GreenXenith no just the colors (I have been making textures and texturepacks for almost 10 years I know how it works XD)

**Comment by GreenXenith** [comment]
Finished chests (for real this time). Working on creative inventory right now.

**Comment by tobyplowy** [comment]
That's great news also the Creative Slots or tabs?

**Comment by GreenXenith** [comment]
Slots.

**Comment by tobyplowy** [comment]
Ok cool

**Comment by GreenXenith** [comment]
On second thought, I'm done for today :P
Got the bg image done.

**Comment by tobyplowy** [comment]
@GreenXenith Ok see you later thanks for your time and have a good day :)

**Comment by D00Med** [comment]
@GreenXenith  bye :)

**Comment by D00Med** [comment]
@tobyplowy  I'm going now, enjoy your day :)

**Comment by DS-Minetest** [comment]
> If you plan on modifying any texture, do NOT change image sizes or slot positions.

Normally you don't set the slot positions in background image. https://github.com/minetest/minetest/blob/master/doc/lua_api.txt#L1582 (You could also add one texture per slot if it's so important to be an image. https://github.com/minetest/minetest/blob/master/doc/lua_api.txt#L1594)

**Comment by GreenXenith** [comment]
The background image slots are aligned with the defined slots.

**Comment by GreenXenith** [comment]
If I do change it to image slots, I will do so later.

**Comment by tobyplowy** [comment]
@GreenXenith any updates?

**Comment by GreenXenith** [comment]
@tobyplowy No :( The beginning of my summer has been a nightmare, and the little time I do get is spend on moderating. I will also be gone for a week. When I get back I will (hopefully) start work on some more stuff.

**Comment by tobyplowy** [comment]
@GreenXenith shuld i add this to the v2 milestone?

**Comment by GreenXenith** [comment]
@tobyplowy Yes.

**Comment by tobyplowy** [comment]
@D00Med @GreenXenith moving to v2

**Comment by GreenXenith** [comment]
I am going to attempt to redo the large chest to use the new opening model

**Comment by tobyplowy** [comment]
@GreenXenith Good :)

**Comment by tobyplowy** [comment]
@GreenXenith maybe try to finish the chests first so you can alredy merge them in to master

**Comment by Wuzzy2** [comment]
Hello. I heard you need help. Please talk to me next week.

**Comment by GreenXenith** [comment]
> Please talk to me next week.

Noted.

**Comment by tobyplowy** [comment]
@GreenXenith easy peasy lemon Squeezy see that wasn't so hard :)

**Comment by tobyplowy** [comment]
@GreenXenith you showed me a working test version of the GUI ones (a while ago)
Is that stable (we can use it for v1 as temporary solution) do you still have that?

**Comment by GreenXenith** [comment]
Definitely not stable.

**Comment by tobyplowy** [comment]
@GreenXenith Oh ok I see

**Comment by GreenXenith** [comment]
I have given up on using the open chest model for the large chest because there is the MT bug that prevents me from doing it easily, and I don't want to figure it out the other way, and it's not that necessary. I will likely end up submitting a PR soon to add the large chest.

**Comment by tobyplowy** [comment]
@GreenXenith ok good 

You have given up on the chests but you are still interested in helping with the inventory GUI right?

**Comment by GreenXenith** [comment]
Of course.

**Comment by tobyplowy** [comment]
@GreenXenith ok thanks :)

**Comment by tobyplowy** [comment]
@GreenXenith soooooo how ja doin XD

**Comment by tobyplowy** [comment]
@GreenXenith when can you work on this again?

**Comment by tobyplowy** [comment]
@GreenXenith do you think you'll be able to add nice GUI for 1.1?

**Comment by GreenXenith** [comment]
@tobyplowy Don't count on it.. Life has been busy for me as of late. Once I can find time, I will try to get the inventory looking spiffy.
